src/TLuaInterpreter.cpp

```diff
@@ -2522,6 +2522,27 @@ int TLuaInterpreter::killTrigger(lua_State* L)
     return 1;
 }
 
+int TLuaInterpreter::remainingTime(lua_State* L)
+{
+    string luaSendText = "";
+    if (!lua_isstring(L, 1)) {
+        lua_pushstring(L, "remainingTime: remainingTime requires a string ID");
+        lua_error(L);
+        return 1;
+    } else {
+        luaSendText = lua_tostring(L, 1);
+    }
+    Host& host = getHostFromLua(L);
+    QString text(luaSendText.c_str());
+    if (host.remainingTime(text) == -1) {
+        lua_pushnil(L);
+        lua_pushstring(L, "timer is inactive or expired");
```

Comment:
Unfortunately you have not trapped the case where the specified timer ***does not exist*** as the `(int) TimerUnit::remainingTime(const QString&)` function/method will also return a -1 if *that* is the situation...

Reply:
Yeah I was going to rewrite that and revert the host class changes at some point today or tomorrow